source/docs/appendices/migration-guides.md

```diff
@@ -42,6 +42,504 @@ In this release we added pagination for orders. In the process, the `orders` Gra
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+`src/web/theme/pages/Product/useProductBySkuLoader.js`
+
+```diff
+-import { useQuery } from "@apollo/react-hooks";
++import { useQuery } from "react-apollo";
+```
+
+</td>
+</tr>
+
+</table>
+
+</details>
+
+- If after the above fixes there is still `@apollo/*` dependencies and you are not sure how to resolve please contact us for support. You can also check [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=8fae8438d46200cbd310b16c30b3078da28b7d42) and [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=418a291f917660b18672b6be2eb2778cb51dfef9) for the changes we did to our codebase.
+
+- Resolvers for fields not in the `schema` now throw errors. To avoid this remove all unneeded `field`/`Type` resolvers that do not have a representation in the GraphQL `schema`. Check our [cleaning commit](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1216/diffs?commit_id=1e2e04e7f28dba2ebd36da50701478a13a58a238) regarding this issue.
+- `makeApolloClientStub` now expects resolvers as an argument instead of mocks. This in in tandem with the [`graphql-tools` upgrade notes](https://www.graphql-tools.com/docs/mocking#migration-from-v7-and-below). Note that we provided an adapter that converts `mocks` to `resolvers` so that old tests are not affected. However it is recommended to make the change. To disable the adapter's warning set `FRONT_COMMERCE_WEB_PRINT_MAKEAPOLLOCLIENTSTUB_MOCKS_DEPRECATION_WARNING` environment variable to `"false"`.
+- The use of `SchemaVisitor`/`SchemaDirectiveVisitor` to implement GraphQL directive is deprecated and it is replaced by the use of the [new API](https://www.graphql-tools.com/docs/schema-directives). We implemented an adapter so that uses of `SchemaVisitor`/`SchemaDirectiveVisitor` will not break, however it is recommended to make the change. Please note that the adapter only supports the use of `visitFieldDefinition` function of the `SchemaVisitor`/`SchemaDirectiveVisitor` class. If you have use for other functions and cannot make the change to the new API please contact us for help.
+- When using node 17 you need to prefix your npm commands by `NODE_OPTIONS=--openssl-legacy-provider` e.g. `NODE_OPTIONS=--openssl-legacy-provider npm start`, `NODE_OPTIONS=--openssl-legacy-provider npm run build`.
+- When using npm >= 7 you need to add `--legacy-peer-deps` to npm install commands. e.g. `npm install --legacy-peer-deps`, `npm i some_dependency --legacy-peer-deps`.
+
+<ul>
+  <li>
+    For good measures reinstall your dependencies after the update:
+    <ul>
+      <li>
+        With npm < 7
+          ```sh
+          rm -rf ./node_modules/
+          npm install
+          ```
+      </li>
+      <li>
+        With npm >= 7
+          ```sh
+          rm -rf ./node_modules/
+          npm install --legacy-peer-deps
+          ```
+      </li>
+    </ul>
+  </li>
+</ul>
```

Comment:
Now we add the problem with dependencies not updating because of the `package-lock.json`, I wonder if we should add `rm package-lock.json` in those instructions (for npm > 8.5~ish)? It doesn't look like a good idea, but just `npm install` doesn't seem to do much now (with or without removing `node_modules`).

Reply:
yes I put the `rm -rf ./node_modules/` which deletes the node modules followed by `npm install`

Reply:
Not the `node_modules`, I was talking about removing the `package-lock.json` directly